### PR TITLE
Use SDK for all input parsing & handling of the parsed data

### PR DESCRIPTION
**Pull request by erdemyerebasmaz**
This PR addresses 
- #508 
and depends on
https://github.com/breez/breez-sdk/pull/123
https://github.com/breez/breez-sdk/issues/118

All `dart_lnurl` references and models are replaced with SDK counterparts. ~Only missing thing is domain information for LNURL payments.~

Includes unit test & integration test changes by @ademar111190.

#### Changelist:
- Remove dart_lnurl and it's usages
  - LNURLParseResult, LNURLWithdrawParams, LNURLPayParams which were used to identify type of lnurl and to generate request data, which is no longer needed as type is known and it's data is already generated from SDK
- Remove domain from lnurlp temporarily, from lnurlw permanently
   - Extract domain from callback url temporarily
   - Domain is not needed for lnurlw according to spec and it was not being used anywhere in the widget tree
- Remove payer data identity LUD-18 related logic
  - https://github.com/breez/breez-sdk/issues/117 is delayed until further notice. 
- Update dependencies to latest
- Improved logging, integration and unit Tests by @ademar111190 in https://github.com/breez/c-breez/pull/507

### CODE REVIEW

File: lib/handlers/input_handler.dart [hunk 1]
```diff
@@ -45,19 +44,21 @@ class InputHandler {
...
+      return handleInvoice(inputState.inputData);
+    } else if (inputType is InputType_LnUrlPay ||
+        inputType is InputType_LnUrlWithdraw ||
+        inputType is InputType_LnUrlAuth ||
+        inputType is InputType_LnUrlError) {
+      return handleLNURL(_context, inputState.inputData);
+    } else if (inputType is InputType_NodeId) {
+      return handleNodeID(inputState.inputData);
+    } else if (inputType is InputType_Url) {
+      return handleWebAddress(inputState.inputData);
```
Comment (erdemyerebasmaz) [inline lib/handlers/input_handler.dart:61]: Was removed with https://github.com/breez/c-breez/commit/c16b2a96c2779a88fbc7241478ef1d559aab67f5

File: lib/bloc/input/input_bloc.dart [hunk 2]
```diff
@@ -46,47 +50,119 @@ class InputBloc extends Cubit<InputState> {
...
-    return InputState(protocol: command.protocol, inputData: invoice);
+  Future<InputState> _handleParsedInput(InputType parsedInput) async {
+    if (parsedInput is InputType_Bolt11 || parsedInput is InputType_BitcoinAddress) {
+      return await handlePaymentRequest(inputData: parsedInput);
+    } else if (parsedInput is InputType_LnUrlPay ||
+        parsedInput is InputType_LnUrlWithdraw ||
+        parsedInput is InputType_LnUrlAuth ||
+        parsedInput is InputType_LnUrlError ||
+        parsedInput is InputType_NodeId ||
+        parsedInput is InputType_Url) {
```
Comment (erdemyerebasmaz) [inline lib/bloc/input/input_bloc.dart:103]: Was removed with [c16b2a9](https://github.com/breez/c-breez/commit/c16b2a96c2779a88fbc7241478ef1d559aab67f5)

File: lib/bloc/input/input_bloc.dart [hunk 3]
```diff
@@ -46,47 +50,119 @@ class InputBloc extends Cubit<InputState> {
...
   }
 
-  Future<InputState?> handlePaymentRequest(String raw, breez_sdk.ParsedInput command) async {
-    final lnInvoice = command.decoded as breez_sdk.LNInvoice;
+  Future<InputState> handlePaymentRequest({required dynamic inputData}) async {
+    late LNInvoice lnInvoice;
+    if (inputData is InputType_Bolt11) {
+      lnInvoice = inputData.invoice;
+    } else if (inputData is InputType_BitcoinAddress) {
+      lnInvoice = await _breezLib.parseInvoice(inputData.address.address);
```
Comment (roeierez) [inline lib/bloc/input/input_bloc.dart:77]: Seems wrong? isn't it a bitcoin address we are trying to parse as invoice?

File: lib/bloc/input/input_bloc.dart [hunk 4]
```diff
@@ -46,47 +50,119 @@ class InputBloc extends Cubit<InputState> {
...
+      amountMsat: lnInvoice.amountMsat ?? 0,
+      expiry: lnInvoice.expiry,
+    );
+
+    return InputState(inputType: InputType_Bolt11, inputData: invoice);
+  }
 
-    return InputState(protocol: command.protocol, inputData: invoice);
+  Future<InputState> _handleParsedInput(InputType parsedInput) async {
+    if (parsedInput is InputType_Bolt11 || parsedInput is InputType_BitcoinAddress) {
```
Comment (roeierez) [inline lib/bloc/input/input_bloc.dart:96]: Looks like it is better to split. Are we doing anything with bitcoin address in breez mobile?

File: test/bloc/input/input_bloc_test.dart [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+      final parsedInput = await breezLib.parseInput(input: input);
+
+      final bloc = InputBloc(breezLib, injector.lightningLinks, injector.device);
+      breezLib.getNodeState();
+
+      expectLater(
+        bloc.stream,
+        emitsInOrder([
+          InputState(isLoading: true),
+          InputState(inputType: parsedInput.runtimeType, inputData: parsedInput),
```
Comment (roeierez) [inline test/bloc/input/input_bloc_test.dart:47]: It seems that we don't need the runtimeType as parameter as we can always calculate it from the inputData

**Review by roeierez — CHANGES_REQUESTED** [review]

File: test/bloc/input/input_bloc_test.dart (hunk 5 above)
Comment (erdemyerebasmaz) [reply test/bloc/input/input_bloc_test.dart]: Yeah, noticed it while testing and was working on it.

Addressed this and other comments here. 117ed79896ed63af3385e95a070e40a098b0097a

**Review by roeierez — APPROVED** [review]

**Pull request merged**